docs/writing-pages.md

```diff
@@ -0,0 +1,186 @@
+Pages should roughly match one-to-one to urls for a web site.  Pages have
+lifecycle methods that are called on them by react-server, which produce the
+html, either on the server or in the browser.  By convention, pages are written
+as [classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes),
+but at their core, pages are Javascript objects with keys named for page
+lifecycle events, and which have corresponding functions that return elements,
+rendered as React instances.
+
+The simplest of pages only needs a `getElements` method
+
+```js
+export default class SimplePage {
+	getElements () {
+		return <h1>Hello react-server</h1>;
+	}
+}
+```
+
+
+# Understanding the page lifecycle
+
+Page lifecycle methods are called in groups that can return asynchronously in
+any order.  We render the head element first
+
+First, we call the following methods together
+
+- renderDebugComments
+- renderTitle
+- renderScripts
+- renderStylesheets
+
+Once `renderStylesheets` has completed, we call the following methods together
+
+- renderMetaTags
+- renderLinkTags
+- renderBaseTag
+
+All of which is then sent to the browser.  Next, we call the rest of the page
+lifecycle is called in order
+
+1. getBodyClasses
+1. getBodyStartContent
+1. getElements
+1. getAboveTheFoldCount (until we've rendered all the above-the-fold content)
+
+Once we reach the above the fold content, we'll start sending javascript.
+
+If you don't provide a page lifecycle method, we provide a "best-guess" default
+value; for instance, you can omit the head methods altogether and get a
+performant and featureful page.
+
+
+# Examples
+
+### Full page
+
+A `react-server` page that serves a full webpage.
+
+```js
+import HttpStatus from 'http-status-codes';
+import MobileEnabled from ('./middleware/MobileEnabled');
+const ExampleComponent = require("./components/example-component");
+const ExampleStore = require("./stores/example-store");
+const exampleAction = require("./actions/example-action");
+
+class ExamplePage {
+	// See [writing middleware](/docs/writing-middleware) for how to write middleware
+	static middleware() { return [MobileEnabled]; }
+
+	handleRoute() {
+		var params = this.getRequest().getQuery();
+		this._exampleStore = new ExampleStore({
+			id: +params.id
+		});
+	}
+
+	getTitle() {
+		return "Example page"
+	}
+
+	getHeadStylesheets() {
+		return [
+			"/styles/example.css",
+			"/styles/reset.css"
+		]
+	}
+
+	getMetaTags() {
+		var tags = [
+			{ name: "example", content: "Demonstrate a full react-server page" },
+		];
+		return tags;
+	}
+
+	getLinkTags() {
+		return [
+			// prefetch analytics to improve performance
+			{ rel: "prefetch", href: "//www.google-analytics.com" },
+		];
+	}
+
+	getBodyClasses() {
+		return ["responsive-page", "typography"];
+	}
+
+	getElements() {
+		return [
+			<ExampleComponent handleOnClick={exampleAction} {...this._exampleStore} />
+		];
```

Comment:
I'd take this out of the array.  Want to steer folks toward `RootContainer` for multiple root elements, and here there's actually only _one_ root element.